### PR TITLE
akeyless-task: new plugin for Akeyless API

### PR DESCRIPTION
**Pull request by benbroadaway**
https://docs.akeyless.io/docs/rest-api

The [existing java library](https://github.com/akeylesslabs/akeyless-java) isn't published to maven central which makes CI a bit tough. It's not particularly special--just uses codegen with an open API spec. So this does the same thing in addition to the main task.

One slight (huge?) different: Concord's client(s) uses swagger codegen which generates code using OkHttp 2.X while codegen generates with OkHttp 3.X. This isn't compatible (as far as I've tested) with concord's targetplatform provided dependencies. So the codegen for this plugin is configured to generate with apache dependencies as a workaround. The bad part is that we lose performance advantage of the included targetplatform dependencies.

Another minor problem is that akeyless isn't clear on the "real" source for their open API spec file. It's included in a number of their open-source repos. But there's no published source that I currently know of which would be nice for staying up-to-date. Here's to hoping they don't break their current API contracts.

The first stab at this plugin is to provide parity with the existing secrets plugins (e.g. Concord, HashiVault). Basically read and write secret strings. If there's a need for other API implementation (e.g. dynamic secrets, other auth methods), it shouldn't be _too_ difficult to add on.

### CODE REVIEW

**Comment by ibodrov** [comment]
Yeah, the okhttp 2.x vs 3.x is a hard problem to deal with without a risk of breaking old plugins.

> The bad part is that we lose performance advantage of the included targetplatform dependencies.
Probably not that critical for this particular use case?

File: tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/SecretExporter.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+public interface SecretExporter {
+    String exportAsString(String o, String n, String p) throws Exception;
```
Comment (ibodrov) [inline tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/SecretExporter.java:24]: Nit: rename parameters to `org`, `secretName` and `password`, perhaps?

File: tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/Util.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * =====
+ */
+
+import org.slf4j.Logger;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class Util {
```
Comment (ibodrov) [inline tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/Util.java:29]: Nit: `final`

**Review by ibodrov — APPROVED** [review]
Some nitpicks. Other than that LGTM.

File: tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/SecretExporter.java (hunk 1 above)
Comment (benbroadaway) [reply tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/SecretExporter.java]: yep yep yep

**Comment by benbroadaway** [comment]
Docs PR: https://github.com/walmartlabs/concord-website/pull/116

**Review by amithkb — APPROVED** [review]

**Pull request merged**